### ISSUE TITLE
Pass already grouped structure to select options

### ISSUE THREAD

**Issue by juristr** [ngx-formly/ngx-formly#1580]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request
```

**Current behavior** / **Expected behavior**
Right for being able to group select items, one has to set the `group` property. Sometimes however, you already have two grouped datasets and you'd like to simply associate them to the options prop. 

so my question, what do you think @aitboudad to allow also to pass in directly a grouped datastructure, like

```
options = [
        {
          label: 'g1',
          group: [
            {
              label: '1',
              value: '1'
            },
            {
              label: '2',
              value: '2'
            }
          ]
        },
        {
          label: 'g2',
          group: [
            {
              label: '3',
              value: '3'
            }
          ]
        }
      ];
```
So basically the transform pipe wouldn't have to do anything, but just pass it along.

If you agree, I can provide a PR.

**Comment by aitboudad** [comment]
make sense to me, that's the exact format we return in case of grouped options

**Comment by juristr** [comment]
> that's the exact format we return in case of grouped options

Yep I know, I looked it up. I think actually this already worked at some point because I found a broken example in my code that used this structure. Which is why I started investigating.

Awesome, I'll provide a PR for this

**Comment by aitboudad** [comment]
I've refactored `SelectOptionsPipe` to make the implementation quite easy:

```patch
private transformOption(option: any, to: ITransformOption) {
  const group = to.groupProp(option);
+  if (Array.isArray(group)) {
+    return {
+      label: to.labelProp(option),
+      group: this.transformOptions(group, to)
+    };
+  }

````

I'll let you do the rest!
